src/main/java/duke/ui/MainWindow.java

```diff
@@ -0,0 +1,73 @@
+package duke.ui;
+
+import duke.Duke;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Duke duke;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    public void setDuke(Duke d) {
+        duke = d;
+    }
+
+    public void showWelcomeMsg() {
+        dialogContainer.getChildren().add(DialogBox.getDukeDialog("Good morning comrade, welcome to KGB.\nWhat can I do for you?", dukeImage));
+    }
+
+    /**
+     * Checks if the userinput is bye, returns a boolean value
+     */
+    public boolean isTerminating() {
+        if (userInput.getText().startsWith("bye")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
+        String input = userInput.getText();
+        String response = duke.getResponse(input);
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(input, userImage),
+                DialogBox.getDukeDialog(response, dukeImage)
```

Comment:
Consecutive line breaks should have an additional 8 white spaces.

Example from CS2103 Java Coding Standard:
```
method(param1,
        object.method()
                .method2(),
        param3);
```
Note the 8 white spaces preceding `object.method()` and 16 white spaces preceding `.method2(),`